ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/displayLanguagePopupServiceAgent.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Inject, Injectable, OnDestroy } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { NotAuthenticatedServiceAgent } from './notAuthenticatedServiceAgent.service';
+import { ConfigurationService } from '../configuration.service';
+import { LoggingService } from '../logging.service';
+import { StudyDetailServiceAgent } from './studyDetailServiceAgent.service';
+import { UserProfile } from '../../models/userProfile';
+import { UserProfileServiceAgent } from './userProfileServiceAgent.service';
+import { BehaviorSubject, Observable, zip } from 'rxjs';
+import { map, skip, take } from 'rxjs/operators';
+import { CompositeDisposable } from '../../compositeDisposable';
+
+@Injectable()
+export class DisplayLanguagePopupServiceAgent extends NotAuthenticatedServiceAgent<boolean> implements OnDestroy {
+  private userDoNotDisplayObsSource: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  private readonly anchor: CompositeDisposable;
+
+  constructor(
+    @Inject('ddp.config') configuration: ConfigurationService,
+    http: HttpClient,
+    logger: LoggingService,
+    private profile: UserProfileServiceAgent,
+    private studyDetailAgent: StudyDetailServiceAgent) {
+    super(configuration, http, logger);
+    this.anchor = new CompositeDisposable();
+  }
+
+  private getStudyDisplayLanguagePopup(): Observable<boolean> {
+    return this.studyDetailAgent.studyDetail.pipe(map(res => res.shouldDisplayLanguageChangePopup));
+  }
+
+  private getUserNotDisplayLanguagePopup(): Observable<boolean> {
+    // Make sure we return the starting value from the profile
+    const sub = this.profile.profile
+        .pipe(take(1))
+        .subscribe(res => this.userDoNotDisplayObsSource.next(res.profile.shouldSkipLanguagePopup));
+    this.anchor.addNew(sub);
```

Comment:
Using this.anchor.addNew(sub); here excessively.
Please leave only take(1). take(1) takes only one value and completes observable, in this case, we don't store subscription.